File: scalafmt-impl/src/scalafmt-0.7/main/java/com/lucidchart/scalafmt/impl/Scalafmtter.java
package com.lucidchart.scalafmt.impl;

import org.scalafmt.Scalafmt;
import org.scalafmt.config.Config;
import org.scalafmt.config.ScalafmtConfig;
import scala.Option;
import scala.collection.immutable.Set$;
import scala.meta.internal.tokenizers.PlatformTokenizerCache;

public final class Scalafmtter implements com.lucidchart.scalafmt.api.Scalafmtter {
    public Object config(String string) {
        return Config.fromHoconString(string, Option.empty()).get();
    }

    // Otherwise, this cache hangs on to a lot
    public void finalize() throws Throwable {
        try {
            PlatformTokenizerCache.megaCache().clear();
        } finally {
            super.finalize();
        }
    }

    public String format(Object config, String code) {
        return Scalafmt.format(code, (ScalafmtConfig)config, Set$.MODULE$.empty()).get();
    }
}

File: scalafmt-impl/src/shared/main/scala/com/lucidchart/scalafmt/impl/Scalafmtter.scala
package com.lucidchart.scalafmt.impl

import com.lucidchart.scalafmt.api
import com.lucidchart.scalafmt.api.Dialect
import java.nio.file.Path
import java.util.function
import org.scalafmt
import org.scalafmt.config.ScalafmtConfig
import scala.meta.dialects

class Scalafmtter(config: ScalafmtConfig) extends api.Scalafmtter {
  self =>

  def formatter(dialect: Dialect) = new function.Function[String, String] {
    private[this] val config = dialect match {
      case Dialect.SBT => scalafmt.Scalafmt.configWithDialect(self.config, dialects.Sbt0137)
      case Dialect.SCALA => self.config
    }
    def apply(code: String) = scalafmt.Scalafmt.format(code, config).get
  }

  def includeFile(file: Path) = config.project.matcher.matches(file.toString)

  // Otherwise, this cache hangs on to a lot
  override protected def finalize() = try ScalametaUtil.clearCache() finally super.finalize()

}
